### PR TITLE
customtimeout plugin

### PR DESCRIPTION
**Pull request by twcollins**

### CODE REVIEW

**Comment by satterly** [comment]
I'm happy to include this plugin but just to make things a bit simpler can we call it "timeout" instead of "customtimeout" ?

File: README.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ Plugins
 
   * [AMQP](plugins/amqp)
   * [Cachet](plugins/cachet)
+  * [Custom Timeout](plugins/customtimeout)
```
Comment (satterly) [inline README.md]:
```
* [Custom Timeout](plugins/timeout)
```

File: plugins/customtimeout/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Configuration
+-------------
+
+Add `customtimeout` to the list of enabled `PLUGINS` in the `alertad.conf` server configuration file and set plugin-specific variables either in the server configuration file or as environment variables.
+
+**Example**
+
+```python
+PLUGINS = ['customtimeout']
```
Comment (satterly) [inline plugins/customtimeout/README.md]:
```
PLUGINS = ['timeout']
```

File: plugins/customtimeout/alerta_customtimeout.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import os
+import logging
+
+from alerta.app import app
+from alerta.plugins import PluginBase
+
+LOG = logging.getLogger('alerta.plugins.customtimeout')
```
Comment (satterly) [inline plugins/customtimeout/alerta_customtimeout.py]:
```
LOG = logging.getLogger('alerta.plugins.timeout')
```

File: plugins/customtimeout/setup.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from setuptools import setup, find_packages
+
+version = '0.0.2'
+
+setup(
+    name="alerta-customtimeout",
```
Comment (satterly) [inline plugins/customtimeout/setup.py]:
```
name="alerta-timeout",
```

File: plugins/customtimeout/setup.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    version=version,
+    description='Alerta plugin to permit a global custom timeout to be supplied via alerta config',
+    url='https://github.com/alerta/alerta-contrib',
+    license='Apache License 2.0',
+    author='Thomas Collins',
+    author_email='thomaswcollins@gmail.com',
+    packages=find_packages(),
+    py_modules=['alerta_customtimeout'],
```
Comment (satterly) [inline plugins/customtimeout/setup.py]:
```
py_modules=['alerta_timeout'],
```

File: plugins/customtimeout/setup.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    install_requires=[],
+    include_package_data=True,
+    zip_safe=True,
+    entry_points={
+        'alerta.plugins': [
+            'customtimeout = alerta_customtimeout:CustomTimeout'
```
Comment (satterly) [inline plugins/customtimeout/setup.py]:
```
'timeout = alerta_timeout:CustomTimeout'
```

File: plugins/customtimeout/setup.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+from setuptools import setup, find_packages
+
+version = '0.0.2'
+
+setup(
+    name="alerta-customtimeout",
+    version=version,
+    description='Alerta plugin to permit a global custom timeout to be supplied via alerta config',
+    url='https://github.com/alerta/alerta-contrib',
+    license='Apache License 2.0',
```
Comment (satterly) [inline plugins/customtimeout/setup.py]: The license of choice for all contributions to this repo is the "MIT" license.

File: plugins/customtimeout/README.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  * The alert 'timeout' attribute is (re)set for each alert to the value specified
+
+
+Installation
+------------
+
+Clone the GitHub repo and run:
+
+    $ python setup.py install
+
```
Comment (satterly) [inline plugins/customtimeout/README.md]: Add ...
```

Or, to install remotely from GitHub run:

    $ pip install git+https://github.com/alerta/alerta-contrib.git#subdirectory=plugins/timeout

Note: If Alerta is installed in a python virtual environment then plugins
need to be installed into the same environment for Alerta to dynamically
discover them.
```

File: plugins/customtimeout/README.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+Configuration
+-------------
+
+Add `customtimeout` to the list of enabled `PLUGINS` in the `alertad.conf` server configuration file and set plugin-specific variables either in the server configuration file or as environment variables.
```
Comment (satterly) [inline plugins/customtimeout/README.md]:
```
Add `timeout` to the list ...
```

**Comment by twcollins** [comment]
@satterly - cheers. 
PR updated with amendments for the changes requested: b4cbf60

**Comment by satterly** [comment]
👍 Thanks.

**Pull request merged**